src/App.js

```diff
@@ -1,21 +1,89 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
-  }
+import React from 'react';
+import {BrowserRouter, Route} from 'react-router-dom'
+import {addIndex, append, curry, evolve, inc, map, pipe, omit, when} from "ramda";
+
+import Toolbar from './Toolbar/Toolbar';
+import Home from './Home/Home';
+import MealDetail from '@/MealDetail/MealDetail';
+import {sortById} from '@/business/meals';
+import MealService from '@/services/meals';
+
+import './App.less';
+
+
+const identifyMeal = (meal, id) => Object.assign({id}, meal);
+
+const unIdentifyMeal = (meal) => omit("id", meal);
+
+const mapWithIndex = addIndex(map);
+
+const useMeal = evolve({
+	lastUsed: () => Date.now(),
+	usedCount: inc
+});
+
+const mealMatchesIndex = curry((index, meal) => meal.id === index);
+
+const updateMealAtIndex = function(index, meals) {
+	let useMealAtIndex = when(mealMatchesIndex(index), useMeal);
+
+	return map(useMealAtIndex, meals);
+};
```

Comment:
In the vue app, i refactored this into a "business/editing" package. there's no reason i couldn't do that here, too. In fact, I could just pull the business package and it's tests from the other repo and drop it in here and start using it, which is a pretty nice feeling and makes me think there might be some merit to this pattern